actor-typed-tests/src/test/scala/org/apache/pekko/actor/typed/SupervisionSpec.scala

```diff
@@ -1133,6 +1161,19 @@ class SupervisionSpec extends ScalaTestWithActorTestKit("""
       }
     }
 
+    "not msg lost on restart multiple times" in new FailingRestartBackoffTestSetup(cntValue = 5) {
+      val exception = new RuntimeException("mockException")
+      val msg: Throw = Throw(exception)
+      val ref = spawn(behv)
+      for (_ <- 1 to cntValue) {
+        ref ! msg
+      }
+      latch.await(1000, TimeUnit.MILLISECONDS)
+      for (_ <- 1 to cntValue) {
+        probe.expectMessage(msg)
+      }
```

Comment:
this test doesn't test the behavior

Reply:
The "StashBuffer" is utilized only during the restart phase and can be cleared once the restart is successful and all messages have been processed. This PR essentially addresses the following issue:

- Reusing the existing "StashBuffer" object during the preparation of a restart to prevent unhandled message loss.
- Resetting the "restartingInProgress" object only after all unhandled messages have been processed following the completion of the restart, thus avoiding the loss of the "StashBuffer".

Reply:
Yes, that's what I learnt from the code.

Reply:
It is that two-issue mean `Restart` and `RestartBackoff` drop an unhandled message,
or "keep `restartingInProgress` before all messages handled" and "refill buffer on preperRestart".

I am struggling with this, and there is another case where, although retaining "restartingInProgress," buffer messages are still lost when encountering an "UnstashException" and the number of retries exceeds`maxRestart`.

Reply:
When the restart time exceeded, drop the buffer seems fine.